### ISSUE TITLE
Option to browse all bookmarks

### ISSUE THREAD

**Issue by IzzySoft** [damko/freedommarks-browser-webextension#18]
I've just migrated my bookmarks from Pocket (a step long overdue), and use this addon now instead. What I much miss is the possibility to browse the list of **all** bookmarks **inside the dropdown** – latest additions first, and gladly "paged" (so I can see e.g. the newest 10 additions and, if needed, can "page back" to older ones step by step). Could you please add that?

Background: My workflow is to browse my feeds on my mobile device while e.g. commuting on the train, mark the posts I need to investigate deeper, and then do the latter on the PC – where it's much more convenient to make notes, copy/paste content etc. So I use a "LIFO" queue ("Last In First Out"), which is what Pocked does fine. Reason for switching is I want to have my data under my own control :wink:

**Comment by damko** [comment]
@IzzySoft I agree all the way but there are some issues with the bookmark app API . Some endpoints are missing and some methods don't have all the features required. Last time I checked the team was working on it. Not sure what's the current status. Pagination was impossible a while ago.

I also plan to investigate the possibility of a side bar where bookmarks can be searched and listed. It should be more comfortable than the popup (which would stay anyway)

Unfortunately I'm really short on time these days (I need to work) so I can't promise any update soon. Sorry.

**Comment by IzzySoft** [comment]
Yes, I saw your checking with the NCB team (but no status update from the other end). I tried an approach (which temporarily worked) – then figured simply clicking the search button (without a term entered) also does the trick. Still, there's a delay until the list is populated in either case.

I think it would be a good idea to keep a local (eg. SQLite) database:

* initially, insert all bookmarks retrieved from NC
* when a single bookmark is added/deleted, process this action on NC and locally
* have a "reload" button. When pressed, the local database is emptied and refilled from NC

All actions like search etc then could take place against the local database, and are much faster. Showing the initial list (as I asked for with this issue) would be without any delay. It would also solve most of your endpoint issues: as you then have all data locally, you can easily check the URLs of all bookmarks (one example you've mentioned there), provided the copy is kept up-to-date by the user.

Unfortunately, I cannot help much with that (my knowledge/experience with FF Extension development is close to zero) and Github no longer lets me open PRs anyway (I use Waterfox browser, which still shows a 56.x version, so Github wants to force me to a newer Firefox – a no-go for me for several reasons which don't belong here) – not even the preview button on this comment works for me.

And thanks for responding in the first place – so at least it's clear this project isn't abandoned. Looking forward to further development! :wink:

**Comment by damko** [comment]
I'm trying to avoid any kind of synchronization because there is always a certain risk of duplication which proportionally grows with the number of devices involved (ex. consider a business scenario with shared bookmarks) so I prefer a simple API client which just performs queries. 

Beside this, I think that the Floccus extension does what you say but uses the browser bookmarks as database. Try it, maybe you like it more.

Yes, I confirm the project is not abandoned :-) just a lil delayed.

**Comment by IzzySoft** [comment]
> there is always a certain risk of duplication

How? Please read my previous comment again. I didn't write "synchronization", but "reload" – i.e. just keep a "local copy", which on reload gets dumped and refilled from Nextcloud (which optionally could also be done after adding a bookmark, if you prefer that). AFAIK that's also how the old Pocket addon worked. I fully agree you shouldn't try "syncing" :wink: The big advantage of the "local copy" is much faster responses.

Well, Floccus *might* work as well for my use case, syncing a specific bookmark folder – which would solve the "speed issue". But then I'd end up with those "read-it-later" items mixed with "permanent bookmarks" when searching via tags, and I'd lose the possibility of having them sorted by e.g. most-recently-added which is essential for me.

So I still hope you reconsider – and, when time permits, implement that (together with sort options :wink:)

**Comment by damko** [comment]
hmm noted. Look I'll come back to you on this as soon as I save some hours for focusing on this

**Comment by IzzySoft** [comment]
I gladly test-drive it (even before publication, via about:debugging) then. Just ping me then.
